prompts/chemprot.py

```diff
@@ -0,0 +1,281 @@
+"""
+Date: 2021.10.10
+
+ChemProt Corpus (https://biocreative.bioinformatics.udel.edu/resources/corpora/chemprot-corpus-biocreative-vi/) prompting annotation
+
+This script does the following:
+
+(1) If the ChemProt corpus is unavailable/unspecified - will download it
+(2) Converts the ChemProt corpus into the BRAT/Standoff format to use pybrat
+(3) Creates the appropriate Example/Relations
+(4) Creates prompts
+
+More details about the dataset can be found in the readme's in the respective datasets.
+
+Overall:
+
+Entities include:
+Chemicals: Chemical entities
+Gene-Y: Gene/protein annotation with a known biological identifier in a DB (as of 2017)
+Gene-N: Gene/protein annotation without a known biological identifier in a database (as of 2017)
+
+Relations Include (exactly quoted from the Readme):
+
+CPR-3: Upregulator, activator, indirect upregulator
+CPR-4: downregulator, inhibitor, indirect downregulator
+CPR-5: agonist, agonist-activator, agonist inhibitor
+CPR-6: antagonist
+CPR-9: sybstrate, product of, substrate product of
+
+**NOTE, the implementation here uses the GOLD STANDARD, as this is canonically evaluated against. There are relations outside these annotations provided but were not group-evaluated (ex: CPR:1, CPR:2, CPR:7, CPR:8, and CPR:10 labels)
+
+Some notes:
+i) The ChemProt dataset is a zipped folder of zipped folders. The individual train/test/sample/dev folders must be also unzipped
+
+ii) The ChemProt dataset needs to be first compiled into standoff format, thus utils_chemprot handles the TSV -> standoff conversion.
+
+Call this script from the head of the biomedical directory
+
+Usage:
+
+python chemprot.py --outdir your/output/dir
+python chemprot.py --outdir your/output/dir --datadir your/chemprot/corpus
+
+"""
+
+import os
+import argparse
+from pathlib import Path
+from loguru import logger
+from functools import partial
+from typing import List, Callable
+from pybrat.parser import BratParser, Example, Relation
+from utils import download, uncompress
+from prompts import DatasetPrompts
+
+from utils_chemprot import chemprot_2_standoff
+
+# Some prompts can be borrowed for template
+from ddi import (
```

Comment:
We'll want to move all the prompt helper functions to their own module at some point. For now this is fine though :)